SourceX/controls/plrctrls.cpp

```diff
@@ -27,75 +27,162 @@ int hsr[3] = { 0, 0, 0 }; // hot spell row counts
 int slot = SLOTXY_INV_FIRST;
 int spbslot = 0;
 
-// 0 = not near, >0 = distance related player 1 coordinates
-coords CheckNearbyObjs(int x, int y, int diff)
+/**
+ * Number of angles to turn to face the coordinate
+ * @param x Tile coordinates
+ * @param y Tile coordinates
+ * @return -1 == down
+ */
+int GetRoteryDistance(int x, int y)
 {
-	int diff_x = abs(plr[myplr]._px - x);
-	int diff_y = abs(plr[myplr]._py - y);
+	int d, d1, d2;
 
-	if (diff_x <= diff && diff_y <= diff) {
-		coords cm = { diff_x, diff_y };
-		//sprintf(tempstr, "N-DIFF X:%i Y:%i", diff_x, diff_y);
-		//NetSendCmdString(1 << myplr, tempstr);
-		return cm;
-	}
-	return { -1, -1 };
+	if (plr[myplr]._px == x && plr[myplr]._py == y)
+		return -1;
+
+	d1 = plr[myplr]._pdir;
+	d2 = GetDirection(plr[myplr]._px, plr[myplr]._py, x, y);
+
+	d = abs(d1 - d2);
+	if (d > 4)
+		return 4 - (d % 4);
+
+	return d;
+}
+
+/**
+ * @brief Get walking distance to cordinate
+ * @param x Tile coordinates
+ * @param y Tile coordinates
+ */
+int GetDistance(int x, int y)
+{
+	char walkpath[25];
+	return FindPath(PosOkPlayer, myplr, plr[myplr]._px, plr[myplr]._py, x, y, walkpath);
+}
+
+/**
+ * @brief Get walking distance to cordinate
+ * @param x Tile coordinates
+ * @param y Tile coordinates
+ */
+int GetDistanceRanged(int x, int y)
+{
+	return path_get_h_cost(plr[myplr]._px, plr[myplr]._py, x, y);
 }
 
-void CheckItemsNearby()
+void FindItemOrObject()
 {
-	for (int i = 0; i < MAXITEMS; i++) {
-		if (CheckNearbyObjs(item[i]._ix, item[i]._iy, 1).x != -1 && item[i]._iSelFlag > 0 && item[i]._itype > -1) {
-			if (dItem[item[i]._ix][item[i]._iy] <= 0)
+	int mx = plr[myplr]._px;
+	int my = plr[myplr]._py;
+	int rotations = 5;
+
+	for (int xx = -1; xx < 2; xx++) {
+		for (int yy = -1; yy < 2; yy++) {
+			if (dItem[mx + xx][my + yy] <= 0)

```

Comment:
Potential OOB?

Reply:
Since this would only happen if the player stands on the edge of the map, and the edge always has a wall tile, do you think it would still need a check here?

Reply:
OK, makes sense!

Reply:
It would be helpful if a comment was added to this code explaining why the check is not needed.